spec/controllers/institutions_controller_spec.rb

```diff
@@ -40,6 +40,51 @@
       expect(user.reload.last_navigation_context).to eq(Institution.last.uuid)
     end
 
+    context "institution invite" do
+      it "create with invite" do
+        invite = PendingInstitutionInvite.make(invited_user_email: user.email)
+        post :create, {institution: {name: invite.institution_name, kind: invite.institution_kind, pending_institution_invite_id: invite.id}}
+
+        institution = Institution.last
+        expect(response).to redirect_to root_path(context: institution.uuid)
+        expect(institution.name).to eq invite.institution_name
+        expect(institution.kind).to eq invite.institution_kind
+
+        invite.reload
+        expect(invite).not_to be_is_pending
+      end
+
+      it "create with invite, overwriting name and type" do
+        invite = PendingInstitutionInvite.make(invited_user_email: user.email)
+        post :create, {institution: {name: "Institution Override", kind: "manufacturer", pending_institution_invite_id: invite.id}}
+
+        institution = Institution.last
+        expect(response).to redirect_to root_path(context: institution.uuid)
+        expect(institution.name).to eq "Institution Override"
+        expect(institution.kind).to eq "manufacturer"
+
+        invite.reload
+        expect(invite).not_to be_is_pending
+      end
+
+      it "create with invite, not invited user" do
+        invite = PendingInstitutionInvite.make
+        post :create, {institution: {name: invite.institution_name, kind: invite.institution_kind, pending_institution_invite_id: invite.id}}
+
+        institution = Institution.last
+        expect(response).to redirect_to root_path(context: institution.uuid)
+
+        invite.reload
+        expect(invite).to be_is_pending
+      end
+
+      it "fails if invite already accepted" do
+        invite = PendingInstitutionInvite.make(invited_user_email: user.email, status: "accepted")
+        expect do
+          post :create, {institution: {name: invite.institution_name, kind: invite.institution_kind, pending_institution_invite_id: invite.id}}
+        end.to raise_error(ActionView::MissingTemplate)
```

Comment:
This error is actually a bit weird and I believe it's a bug that this happens.

When there is no matching invite, the controller calls `#no_data_allowed` which is supposed to redirect. But the redirect is guarded by a restriction that seems to prevent it from executing. As a result, the `new_from_invite_data` falls through and rails tries to render the respective template (which doesn't exist).

Reply:
This is tracked in #1467